esp/public/media/scripts/program/modules/adminclass.js

```diff
@@ -187,105 +187,119 @@ function fillClasses(data)
     var teacher_list_string = teacher_list.join(", ");
+    return $j("<td class='clsleft classname' style='font-style: italic'> \
+      <span title='Teacher Names'> \
+        " + teacher_list_string + " \
+      </span> \
+    </td>");
+}
 
-    var section_link_list = "";
-    for (var i = 0; i < clsObj.sections.length; i++)
-    {
-	var section = sections[clsObj.sections[i]];
-	section_link_list = section_link_list.concat("<a href='/teach/"+base_url+"/select_students/"+section.id+"'>Sec. "+section.index+"</a><br />");
-    }
-    
-    var class_title_trimmed = clsObj.title;
-    if (class_title_trimmed.length > 40)
-    {
-	class_title_trimmed = class_title_trimmed.substring(0, 40);
-	class_title_trimmed = class_title_trimmed.concat("...");
-    }
+function createManageButtonTd(clsObj) {
+    return $j("<td class='clsmiddle'> \
+       <form method='post' action='/manage/" + base_url + "/deleteclass/" + clsObj.id + "' onsubmit='return deleteClass();'> \
+         <input class='button' type='submit' value='Delete' /> \
+       </form> \
+    </td>");
+}
 
-    var status_details = getStatusDetails(clsObj.status);
+function createDeleteButtonTd(clsObj) {
+    return $j("<td class='clsmiddle'> \
+       <form method='post' action='/manage/" + base_url + "/deleteclass/" + clsObj.id + "' onsubmit='return deleteClass();'> \
+         <input class='button' type='submit' value='Delete' /> \
+       </form> \
+    </td>");
+}
 
-    template = template.replace(new RegExp("{{ cls.id }}", "g"), clsObj.id)
-	.replace(new RegExp("{{ cls.title }}", "g"), class_title_trimmed)
-	.replace(new RegExp("{{ cls.emailcode }}", "g"), clsObj.emailcode)
-	.replace(new RegExp("{{ teacher_names }}", "g"), teacher_list_string)
-	.replace(new RegExp("{{ section_links }}", "g"), section_link_list)
-	.replace(new RegExp("{{ program.getUrlBase }}", "g"), base_url)
-	.replace(new RegExp("{{ title_css_class }}", "g"), status_details.classes.join(" "))
-	.replace(new RegExp("{{ cls_status }}", "g"), status_details['text']);
+function createLinkButtonTd(clsObj, shortTitle, type, verb) {
+    return $j("<td class='clsmiddle'> \
+      <a href='/manage/" + base_url + "/" + type + "/" + clsObj.id + "' title='" + verb + " " + shortTitle + "' \
+       class='abutton' style='white-space: nowrap;'>" + verb + "</a> \
+    </td>");
+}
+function createStatusButtonTd(clsObj, shortTitle) {
+    return $j("<td class='clsmiddle rapid_approval_button'> \
+      <a href='#' title='Set the status of " + shortTitle + "' class='abutton' style='white-space: nowrap;' onclick='show_approve_class_popup(" + clsObj.id + "); return false;'>Status</a> \
+    </td>");
+}
+
+function createClassRow(clsObj)
+{
+    var shortTitle = clsObj.title;
+    if (shortTitle.length > 40) {
+        shortTitle = shortTitle.substring(0, 40);
+        shortTitle = shortTitle.concat("...");
+    }
 
-    // Turn the template into a jQuery node
-    $node = $j(template);
+    var tr = $j(document.createElement('tr'));
```

Comment:
Previously I think this was using the convention that variables starting with `$` are jQuery nodes.  I'm not super attached to that but if you feel like it it might be nice.